sklego/meta/hierarchical_predictor.py

```diff
@@ -281,12 +289,24 @@ def fit(self, X, y=None):
         if self.n_features_in_ < 1:
             msg = "Found 0 features, while a minimum of 1 if required."
             raise ValueError(msg)
+        
+        self.has_sw_ = sample_weight is not None
+        
+        if self.has_sw_ and not self.estimator_supports_sample_weight:
+            msg = f"Estimator does not support sample_weight."
+            raise ValueError(msg)
+        sample_weight = _check_sample_weight(sample_weight, X, None, ensure_non_negative=True)
```

Comment:
Here you probably need to cast X to native since scikit-learn does not (yet) work with narwhals objects, also `dtype` argument has already None as default:

```suggestion
        sample_weight = _check_sample_weight(sample_weight, X.to_native(), ensure_non_negative=True)
```